packages/@icon-magic/config-reader/src/schemas/config-schema.ts

```diff
@@ -3,14 +3,61 @@ const variantProperties = {
   path: { type: 'string', minLength: 1 }
 };
 
+const pluginProperties = {
+  name: { type: 'string' },
+  iterants: {
+    type: ['array'],
+    items: {
+      type: 'string'
+    }
+  },
+  params: ['object'],
+  writeToOutput: {
+    type: 'boolean',
+    default: true
+  }
+};
+
 const buildConfigProperties = {
-  outputPath: { type: ['string', 'null'] },
-  plugins: { type: ['array', 'null'] }
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
+  },
+  plugins: {
+    type: ['array', 'null'],
+    items: {
+      type: ['object'],
+      properties: pluginProperties,
+      additionalProperties: true
+    }
+  }
 };
 
 const distributeConfigProperties = {};
 
-const generateConfigProperties = {};
+const generateConfigProperties = {
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
+  },
+  types: {
+    type: ['array'],
+    items: {
+      type: ['object'],
+      properties: {
+        name: { type: 'string' },
```

Comment:
I'm wondering if we should check for 'svg', 'png', and 'webp' here since those are the only supported types so far?

Reply:
you mean 'svg' and 'raster'? 'svg', 'png', and 'webp' are for the Flavor types I believe

Reply:
Yes, 'svg' and 'raster' :)